File: Chapter07/VB/VBProject_07/ASPAJAX/2-ASPUserControl/FunBadge.Web/FunBadge.ascx.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Drawing

' Namespace FunBadge.Web

    Public Partial Class SilverlightControl
        Inherits System.Web.UI.UserControl
        Private _startValue As Integer = 0
        Private _backgroundColor As String = "White"

        Public Property StartValue() As Integer
            Get
                return _startValue
            End Get

            Set
                _startValue = value
            End Set
        End Property
        Public Property BackgroundColor() As String
            Get
                return _backgroundColor
            End Get

            Set
                _backgroundColor = value
            End Set
        End Property
        protected Sub Page_Load(sender As Object, e As EventArgs)

            Silverlight1.Source = "ClientBin/FunBadge.xap"
            Silverlight1.InitParameters = "StartValue=" +_startValue.ToString() + ",BackgroundColor=" +_backgroundColor
        End Sub '   Page_Load
    End Class   '   SilverlightControl
' End Namespace   '   FunBadge.Web
' ..\Project_07\ASPAJAX\2-ASPUserControl\FunBadge.Web\FunBadge.ascx.cs

File: Chapter07/C#/Project/ASPAJAX/2-ASPUserControl/FunBadge.Web/FunBadge.ascx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Drawing;

namespace FunBadge.Web
{
    public partial class SilverlightControl : System.Web.UI.UserControl
    {
        private int _startValue =0;
        private string _backgroundColor = "White";
       
        public int StartValue
        {
            get { return _startValue; }
            set { _startValue = value; }
        }
        public string BackgroundColor
        {
            get { return _backgroundColor; }
            set { _backgroundColor = value; }
        }
        protected void Page_Load(object sender, EventArgs e)
        {
            Silverlight1.Source = "ClientBin/FunBadge.xap";
            Silverlight1.InitParameters = "StartValue=" +_startValue.ToString() + ",BackgroundColor=" +_backgroundColor;
        }
    }
}